src/protocol/tx_format/message_pointer.md

```diff
@@ -0,0 +1,9 @@
+# MessagePointer
+
+The location of the message on the base chain. It can be used by UTXOs as a reference to the message or by the transaction itself to make it unique.
+
+| name          | type     | description        |
+|---------------|----------|--------------------|
+| `blockHeight` | `uint32` | Block height.      |
+| `txIndex`     | `uint16` | Transaction index. |
+| `logIndex`    | `uint16` | Log index.         |
```

Comment:
This is fairly EVM specific, is this really necessary if we already have a nonce?

Reply:
It needs to serve the same purpose as the txPointer, which I assume is for helping in transaction validation when a block is published? If that's the case, shouldn't it be EVM specific? The nonce doesn't currently help all that much because `nonce` is not indexed on the Ethereum event. We could make it indexed, but I also don't think we should be assuming Ethereum logs will never be pruned from full Ethereum nodes. I know there are several EIPs on the roadmap for log pruning to help ease state growth concerns

Reply:
I see, is it right to assume we don't store any message metadata on the bridging contract itself and it just does the log emitting?

Would it be viable to just reuse the nonce field as a concatenation of these fields to lighten our tx payload size?

Reply:
Correct, nothing is stored only an event is emitted to save gas. Using the nonce wouldn't lessen the payload size since it's a `uint64` (vs a `uint32` + `uint16` + `uint16`)

Reply:
Would we still need the nonce on L2 if we have the pointer?

Reply:
You mean remove the nonce from `InputMessage`? I guess it's not necessary since it's implicitly part of the messageId and it's not like it helps the client query Ethereum for the message in question. 

Side note: is there a way for newly syncing nodes to quickly download and trust indexed data about messages from other nodes? Is there a way to do this for the UTXO set in order to quickly catch up? I'm not sure how this could be done since our header doesn't include any state items like that.

Reply:
> You mean remove the nonce from InputMessage?

Yep, as it seems like the message pointer already serves as a nonce. No need to have two nonces.

> is there a way for newly syncing nodes to quickly download and trust indexed data about messages from other nodes?

This is something we could enable a toggle option for, but it means trusting the sequencer. Potentially useful for infura type services but I wouldn't advise this as a default when running a full-node for any high-risk off-chain bridging (e.g. centralized custodial services).

Reply:
closing this thread. I think keeping the nonce is fine as replacing it with message pointer may not be as straightforward and would be a larger breaking change (e.g. message id and signing procedures would need to have breaking changes)